pandas/core/series.py

```diff
@@ -2644,6 +2645,8 @@ def combine(self, other, func, fill_value=None):
 
         if is_categorical_dtype(self.values):
             pass
+        elif is_bool(new_values[0]) and not is_sparse(self.values):
```

Comment:
hmm, what test break prompted this?

Reply:
All of the integer's `TestComparisonOps.test_compare_array` call `Series.combine` to get the expected results, which are boolean and would be converted to int without checking for `is_bool`. But if the original array is sparse, then if we skip `is_sparse` here the expected array would not become sparse.

Reply:
I would rather change the test to do this properly that put this in here. pls remove.

Reply:
Well it's embarrassing but I haven't noticed that there are two variants of `test_integer`
here https://github.com/pandas-dev/pandas/blob/cb207aac2179075f092fd068176ac48314a5c951/pandas/tests/arrays/test_integer.py#L332 and here https://github.com/pandas-dev/pandas/blob/cb207aac2179075f092fd068176ac48314a5c951/pandas/tests/extension/test_integer.py#L152. The former overrides `_compare_other` function and doesn't use `Series.combine` and thus doesn't fail, while the latter fails because is calls parent's `_compare_other`. Do you think it's ok to import this test from arrays/test_integer.py into extension/test_integer.py?

Reply:
hmm that's prob an error in the original splitting. The idea is that the extension/* test the public interface and should use the base imports; while thie arrays/* shouldn't be using this at all.

Are the tests the same?

Reply:
Looks like it. The way I see it, the main diff for this particular case is that the one in extensions computes expected as `expected = s.combine(other, op)` and the one in arrays as `expected = pd.Series(op(data._data, other))`

Reply:
@jreback, any advice on this one?

Reply:
these should be using the same mechnaism as in the arrays which uses ``pd.Series(op(data._data, other))``. you can update these from using .combine and I think this would be ok

Reply:
I've found this PR #21183, so it seems `s.combine` use is expected to work too.

Reply:
@jreback, did you see my last comment? It's seems some people need that s.combine works, so should we update to op(data._data, other) regardless?

Reply:
I don't like this special case that was added. so probably needs some more investigation. I would either investigate more here or xfail that test.